Wine 10.0 + update all outdated dependencies and a few cleanups (ready to merge)

user1-github opened the pull request:
Before the release of Wine 10.0, I'd first like to see if everything builds with all my other changes. When Wine 10 is released, I'll update this MR and it'll be ready to merge.

user1-github committed:
Update FAudio to 25.01

user1-github committed:
Update vkd3d to 1.14

user1-github committed:
Update winetricks to 20250102

user1-github committed:
Update spirv-headers to 1.3.296.0

user1-github committed:
Update openldap to 2.6.9

user1-github committed:
Update perl-JSON to 4.10

user1-github committed:
Update rpcsvc-proto to 1.4.4

user1-github committed:
Update unixODBC to 2.3.12

user1-github committed:
Update cabextract to 1.11

user1-github committed:
Make sure Wine builds with Wayland driver

user1-github committed:
Update samba to 4.21.3

user1-github committed:
Update wine-mono to 9.4.0

user1-github committed:
Downgrade to samba 4.20.6

I can't figure out why samba 4.21.x fails to build, so try with the previous version.

user1-github commented:
Btw, I heard Wine Wayland driver is not in experimental state anymore in Wine 10. So maybe now it's a good time to replace --socket=x11 with --socket=fallback-x11?

On a different note, I noticed that since a few months ago Wine Flatpak is not displayed in Flathub search results (it previously used to be).

Erick555 commented:
If wine doesn't use wayland by default for everything then it should stay as is.

user1-github commented:
I just checked what folks who recently tried the Wayland driver say about it, and it still has some papercuts. So yeah, better leave it as is.

user1-github committed:
Remove FAudio

The native Linux version of FAudio used to be a dependency of Wine for a brief period around 2020, but [it's no longer needed](https://forum.winehq.org/viewtopic.php?t=32192) starting with Wine 6.21 and since at least 2021, Wine [now bundles its own version of FAudio](https://bugs.winehq.org/show_bug.cgi?id=51921#c4) in PE format.

user1-github committed:
Remove vkd3d + minor cleanup

Like FAudio, vkd3d is now also bundled with Wine since 2022 as can be seen in the main Wine source repo: https://gitlab.winehq.org/wine/wine/-/commits/master/libs/vkd3d?ref_type=heads .

The HAL library was deprecated a long time ago and removed with Wine 7.0, so --without-hal is no longer a valid config option.

user1-github committed:
Remove openldap + try updating to the latest samba again

It turns out openldap is also not needed anymore. Bundled with Wine since 7.22: https://www.winehq.org/announce/7.22 .

Adding --without-ldb-lmdb to samba config opts is likely the only way to make the latest version build.

user1-github committed:
Add openldap back

Turns out samba needs it.

user1-github committed:
Revert back to samba 4.20.6 again

Now it failed for a different reason which I don't understand.

user1-github committed:
Update to Wine 10.0

user1-github committed:
Add Wine 10.0 to appdata

user1-github commented:
bot, build

user1-github commented:
@bbhtt Just tested the new build with Wine 10.0, works fine. This can now be merged.

Erick555 commented:
@user1-github thx for working in it. Do you know anything abut [wow64](https://github.com/flathub/org.winehq.Wine/pull/39) status in recent wine?

user1-github commented:
> @user1-github thx for working in it. Do you know anything abut [wow64](https://github.com/flathub/org.winehq.Wine/pull/39) status in recent wine?

My pleasure. I can see that the last news regarding wow64 were with the release of [Wine 9.0](https://gitlab.winehq.org/wine/wine/-/releases/wine-9.0) - it was completed in that version, but 32 bit OpenGL apps that run in that mode will have reduced performance and 16 bit apps won't work.
Wine 10.0 release notes don't mention anything new regarding wow64.

I actually thought about the idea of reviving that wow64 branch from the original maintainer, but considering its current limitations (mainly the reduced OpenGL performance), idk if it's practical right now.

On a different note, I wonder if bundling Samba with this Flatpak is actually necessary. I mean I'm sure there are other ways to make the Flatpak access samba shares. Maybe there's a specific Flatpak permission for that?

Erick555 commented:
I wonder if it can use `--enable-archs=x86_64,i386` but keep all current 32 libs. Not sure what will be used then and if there's a way to cycle between native 32bit and wow64.

> On a different note, I wonder if bundling Samba with this Flatpak is actually necessary. I mean I'm sure there are other ways to make the Flatpak access samba shares. Maybe there's a specific Flatpak permission for that?

it's not for accessing samba shares but for running ms office under wine which refuses to work without winbind (part of samba), see https://github.com/flathub/org.winehq.Wine/issues/3. Actually ms office stopped working for me even with samba since couple of months but didn't had time to debug it.

user1-github commented:
> I wonder if it can use `--enable-archs=x86_64,i386` but keep all current 32 libs. Not sure what will be used then and if there's a way to cycle between native 32bit and wow64.

I'm not sure what you mean, but the new wow64 mode doesn't use any native (unix) 32 bit libs. One of the main reasons for making the new wow64 mode is to run 32 bit software on systems where there are no native 32 bit libs (like recent MacOS versions). So I don't think you can cycle between the 2 modes.

Btw, I just noticed that interestingly, the official WineHQ packages for Fedora 40 and higher [are built in the new wow64 mode exclusively](https://gitlab.winehq.org/wine/wine/-/wikis/Fedora). For Debian and Ubuntu however, they continue to build separate 64 and 32 bit Wine packages.

Erick555 commented:
I mean if we enable this option but don't remove any flatpak 32bit content.

user1-github commented:
> I mean if we enable this option but don't remove any flatpak 32bit content.

Oh, then I guess Wine itself will just not touch any 32 bit stuff. If you mean building Wine in wow64 mode, but also keeping the regular 32 bit Wine, then idk how will that work out or if it's even possible.
But why would you want to do this?

Erick555 commented:
To test wow64 without enforcing it

user1-github commented:
On another note, Wine [now has a MR](https://gitlab.winehq.org/wine/wine/-/merge_requests/7226) to support the new NTSYNC kernel driver. So next year's Wine 11.0 should utilise NTSYNC if you have kernel 6.14+.
I know that Flatpak can restrict access to some kernel features (for example, Chromium browsers can't use namespaces in a Flatpak environment), so I wonder if Wine Flatpak would be able to utilise NTSYNC.

Erick555 commented:
namespaces are blocked due to security. As I understand `/dev/ntsync` access is needed for ntsync feature and wine has `--device=all` permission which grants access.

user1-github commented:
Btw, when I did my last pull request yesterday, I've noticed that Wine mono is provided as a .tar.xz archive in the manifest, but Wine gecko for some reason is provided as an .msi installer.
I thought the whole point of the shared mono and gecko installations is to avoid installing them into the Wine prefix, so that it will avoid duplication if someone has more than one prefix. Providing Wine gecko as an .msi installer seems to kill that point.
Maybe you know the reason behind it?

Erick555 commented:
I don't know. Looks like omission. Maybe it was copied from somewhere and the source still used .msi for gecko back then. The first version in [flatpak](https://github.com/flathub/org.winehq.Wine/commit/be634e2b1a20edc6f6bf20a3e4c554757e74962f#diff-bcc754adaf8f3b31abc705f27f66211e92e24380f091b6b895822cf514c25d84) was 2.47.2 and it was also the first version than had tar.xz available so maybe not everyone switched to it at that time.

user1-github commented:
Then I'll probably change it to use the tarball. From reading in different sources, everyone recommend to use the .tar.xz for the shared install.
